pkg/controller/unreachable/unreachable_controller_test.go

```diff
@@ -0,0 +1,239 @@
+				if cond.Status != corev1.ConditionTrue {
+					t.Errorf("Did not get expected state for unreachable condition. Expected: \n%#v\nGot: \n%#v", corev1.ConditionTrue, cond.Status)
+				}
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			fakeClient := fake.NewFakeClient(test.existing...)
+			rcd := &ReconcileRemoteMachineSet{
+				Client:                        fakeClient,
+				scheme:                        scheme.Scheme,
+				logger:                        log.WithField("controller", "unreachable"),
+				remoteClusterAPIClientBuilder: testRemoteClusterAPIClientBuilder,
+			}
+
+			namespacedName := types.NamespacedName{
+				Name:      testName,
+				Namespace: testNamespace,
+			}
+
+			_, err := rcd.Reconcile(reconcile.Request{NamespacedName: namespacedName})
+
+			if test.validate != nil {
+				cd := &hivev1.ClusterDeployment{}
+				err := fakeClient.Get(context.TODO(), namespacedName, cd)
+				if err != nil {
+					t.Errorf("unexpected error: %v", err)
+					return
+				}
+				test.validate(t, cd)
+			}
+
+			if err != nil && !test.expectError {
+				t.Errorf("Unexpected error: %v", err)
+			}
+			if err == nil && test.expectError {
+				t.Errorf("Expected error but got none")
+			}
+		})
+	}
+}
+
+func TestIsTimeForConnectivityCheck(t *testing.T) {
+	cd := addFalseUnreachableCondition()
+	cond := controllerutils.FindClusterDeploymentCondition(cd.Status.Conditions, hivev1.UnreachableCondition)
+	if !isTimeForConnectivityCheck(cond) {
+		t.Error("Exppected isTimeForConnectivityCheck() to return true")
+	}
+}
+
+func testRemoteClusterAPIClientBuilder(secretData string) (client.Client, error) {
+	err := errors.New("cluster not reachable")
+	return nil, err
+}
+
+func testClusterDeployment() *hivev1.ClusterDeployment {
+	cd := &hivev1.ClusterDeployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:       testName,
+			Namespace:  testNamespace,
+			Finalizers: []string{hivev1.FinalizerDeprovision},
+			UID:        types.UID("1234"),
+		},
+		Spec: hivev1.ClusterDeploymentSpec{
+			ClusterName: testClusterName,
+			SSHKey: &corev1.LocalObjectReference{
+				Name: sshKeySecret,
+			},
+			ControlPlane: hivev1.MachinePool{},
+			Compute:      []hivev1.MachinePool{},
+			PullSecret: corev1.LocalObjectReference{
+				Name: pullSecretSecret,
+			},
+			Platform: hivev1.Platform{
+				AWS: &hivev1.AWSPlatform{
+					Region: "us-east-1",
+				},
+			},
+			Networking: hivev1.Networking{
+				Type: hivev1.NetworkTypeOpenshiftSDN,
+			},
+			PlatformSecrets: hivev1.PlatformSecrets{
+				AWS: &hivev1.AWSPlatformSecrets{
+					Credentials: corev1.LocalObjectReference{
+						Name: "aws-credentials",
+					},
+				},
+			},
+		},
+		Status: hivev1.ClusterDeploymentStatus{
+			ClusterID: testClusterID,
+			AdminKubeconfigSecret: corev1.LocalObjectReference{
+				Name: "kubeconfig-secret",
+			},
+		},
+	}
+	cd.Status.Installed = true
+	return cd
+}
+
+func addFalseUnreachableCondition() *hivev1.ClusterDeployment {
```

Comment:
testClusterDeploymentWithUnreachableCondition, should probably accept status true/false as well as an argument.